usr/share/rear/skel/default/etc/scripts/system-setup

```diff
@@ -175,26 +180,34 @@ echo -e "\nRelax-and-Recover rescue system is ready\n"
 # unless in unattended_recovery mode where there is normally no watching user:
 unattended_recovery || sleep $USER_INPUT_UNATTENDED_TIMEOUT
 
-# In debug mode run the automated 'rear recover' also with debug options.
+# Make it explicit that the 'recover' workflow is always verbose (cf. usr/sbin/rear)
+# so when a 'rear $rear_options recover' command is shown to the user it contains '-v':
+rear_options='-v'
+# In debug mode run an automated 'rear $rear_options recover' command in debugscript mode.
 # Because the kernel command line option 'debug' means 'set -x' for the system setup scripts
-# it also means '-D' (i.e. 'set -x') for the automated 'rear recover' run:
-if rear_debug ; then
-    rear_debug_options='-D'
-else
-    rear_debug_options=''
-fi
+# it should also mean '-D' (i.e. 'set -x') for an automated 'rear $rear_options recover' run:
+rear_debug && rear_options='-D'
+# In unattended_recovery mode run an automated 'rear $rear_options recover' in non-interactive mode:
+unattended_recovery && rear_options+=' --non-interactive'
 
-# Launch "rear recover" automatically:
+# In automatic_recovery mode call RECOVERY_COMMANDS automatically
+# but without automated calling REBOOT_COMMANDS after successful recovery:
 if automatic_recovery ; then
     choices=( "View Relax-and-Recover log file(s)"
-              "Go to Relax-and-Recover shell"
+              "Login at the rescue system"
             )
-    echo -e "\nLaunching 'rear recover' automatically\n"
-    if rear $rear_debug_options recover ; then
-        echo -e "\n'rear recover' finished successfully\n"
+    eval echo "Launching \'$RECOVERY_COMMANDS_LABEL\' automatically"
+    for command in "${RECOVERY_COMMANDS[@]}" ; do
+        rear_debug && eval echo "Running RECOVERY_COMMANDS \'$command\'"
+        eval "$command"
+        recovery_command_exit_code=$?
```

Comment:
Isn't it better to exit the loop after the first failed command, with its exit code, instead of continuing with the next commands in the loop? (I am not sure myself, but it looks safer to stop early if there is a problem.)

Reply:
I don't know if we stop executing the commands after first failure.

I documented in default.conf exactly how it is executed
so the user knows that implementation detail
because this implementation detail is important
for the user to understand how it behaves.

I did not stop executing the commands after first failure
because I liked to keep the default 'for' loop semantics
which is according to "man bash"
```
The return status is the exit status
of the last command that executes.
```
I think I had even tested that a 'break' command
as one of the RECOVERY_COMMANDS should exit
the 'for' loop so it should be possible
to exit RECOVERY_COMMANDS early when needed.
At least on command line this works:
```
# COMMANDS=( 'echo begin' 'break' 'echo end' )

# ( set -x ; for command in "${COMMANDS[@]}" ; do eval "$command" ; done ; echo $? )
+ for command in "${COMMANDS[@]}"
+ eval 'echo begin'
++ echo begin
begin
+ for command in "${COMMANDS[@]}"
+ eval break
++ break
+ echo 0
0

# COMMANDS=( 'echo begin' 'break 0 2>/dev/null' 'echo end' )

# ( set -x ; for command in "${COMMANDS[@]}" ; do eval "$command" ; done ; echo $? )
+ for command in "${COMMANDS[@]}"
+ eval 'echo begin'
++ echo begin
begin
+ for command in "${COMMANDS[@]}"
+ eval 'break 0 2>/dev/null'
++ break 0
+ echo 1
1
```

Reply:
I would keep this logic simple and straight-forward, meaning that the user should add their own error handling or control logic if needed.

Can't they use the ReaR functions, too? e.g. `some_command || Error "a problem happened"` and then this is good enough for me.

Reply:
One more why I think we should not stop executing at first failure:

How would it help the user if we did that during recovery?

When some command in RECOVERY_COMMANDS fails
but the whole recovery (somehow) succeeds nevertheless
all could be (sufficiently) OK for the user.

In contrast when any command in RECOVERY_COMMANDS fails
and we abort the whole recovery we enforce a hardcoded
stop onto the user who then has to manualy do something
to get recovery running.
If that failed command was not crucial for recovery
the user could be rather annoyed.

Reply:
Ony FYI a side note:
I find it questionable why diskrestore.sh is run with 'set -e'.
But I think for diskrestore.sh it is less problematic because
diskrestore.sh contains our generated code which should work
and if not it is almost certainly a real severe problem.